account_invoice_change_currency/tests/test_account_invoice_change_currency.py

```diff
@@ -0,0 +1,334 @@
+                    login="fm",
+                    email="accountmanager@yourcompany.com",
+                    groups_id=[(6, 0, [res_users_account_manager.id])],
+                )
+            )
+        )
+
+        # Needed to create invoice
+        self.account_type1 = self.env["account.account.type"].create(
+            {
+                "name": "acc type test 1",
+                "type": "receivable",
+                "include_initial_balance": True,
+                "internal_group": "asset",
+            }
+        )
+        self.account_type2 = self.env["account.account.type"].create(
+            {
+                "name": "acc type test 2",
+                "type": "other",
+                "include_initial_balance": True,
+                "internal_group": "asset",
+            }
+        )
+        self.account_account = self.env["account.account"].create(
+            {
+                "name": "acc test",
+                "code": "X2020",
+                "user_type_id": self.account_type1.id,
+                "reconcile": True,
+            }
+        )
+        self.account_account_line = self.env["account.account"].create(
+            {
+                "name": "acc inv line test",
+                "code": "X2021",
+                "user_type_id": self.account_type2.id,
+                "reconcile": True,
+            }
+        )
+        self.product_1 = self.env["product.product"].create({"name": "Product 1"})
+        self.product_2 = self.env["product.product"].create({"name": "Product 2"})
+        self.analytic_account = self.env["account.analytic.account"].create(
+            {"name": "test account"}
+        )
+        self.tax_account = self.env["account.account"].search(
+            [
+                (
+                    "user_type_id",
+                    "=",
+                    self.env.ref("account.data_account_type_current_assets").id,
+                )
+            ],
+            limit=1,
+        )
+
+    def create_journal_type(self, type_string=None):
+        if not type_string:
+            type_string = "purchase"
+        sequence = self.env["ir.sequence"].create(
+            {
+                "name": "Journal",
+                "prefix": type_string,
+                "padding": 6,
+                "company_id": self.env.ref("base.main_company").id,
+            }
+        )
+        account_journal_type = self.env["account.journal"].create(
+            {
+                "name": "Journal",
+                "code": type_string,
+                "type": type_string,
+                "sequence": sequence.id,
+            }
+        )
+        return account_journal_type
+
+    def create_simple_invoice(
+        self, date=False, context=None, inv_type=None, journal=None
+    ):
+        if not context:
+            context = {}
+        if not journal:
+            journal = self.create_journal_type(type_string="purchase")
```

Comment:
No need to create a new journal every time an invoice is created, you could create a new one in the `setUp` method

Reply:
Changes that I applied makes this code unnecessary